### ISSUE TITLE
[flutter_tools] XmlTagException during flutter run

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#106365]
on `beta/3.1.0-9.0.pre`
with command: `flutter --no-color run --machine --track-widget-creation --device-id=emulator-5554 --start-paused --dart-define=flutter.inspector.structuredErrors=true lib\main.dart`

```
XmlTagException: Expected </application>, but found </activity> at 37:4
at XmlAnnotator.annotate(annotator.dart:88)
at XmlEventIterator.moveNext(iterator.dart:32)
at Iterable.forEach(iterable.dart:325)
at XmlNodeDecoder.convertIterable(node_decoder.dart:53)
at new XmlDocument.parse(document.dart:34)
at AndroidProject.computeEmbeddingVersion(project.dart:649)
at AndroidProject.checkForDeprecation(project.dart:592)
at FlutterProject.checkForDeprecation(project.dart:381)
at FlutterCommand.verifyThenRunCommand(flutter_command.dart:1319)
at <asynchronous gap>(async)
at FlutterCommand.run.<anonymous closure>(flutter_command.dart:1195)
at <asynchronous gap>(async)
at AppContext.run.<anonymous closure>(context.dart:150)
at <asynchronous gap>(async)
at CommandRunner.runCommand(command_runner.dart:209)
at <asynchronous gap>(async)
at FlutterCommandRunner.runCommand.<anonymous closure>(flutter_command_runner.dart:281)
at <asynchronous gap>(async)
at AppContext.run.<anonymous closure>(context.dart:150)
at <asynchronous gap>(async)
at FlutterCommandRunner.runCommand(flutter_command_runner.dart:229)
at <asynchronous gap>(async)
at run.<anonymous closure>.<anonymous closure>(runner.dart:62)
at <asynchronous gap>(async)
at AppContext.run.<anonymous closure>(context.dart:150)
at <asynchronous gap>(async)
at main(executable.dart:94)
at <asynchronous gap>(async)
```

**Comment by christopherfujino** [comment]
The fix for this missed the beta 3.3.0-0.0.pre branch cut

**Comment by christopherfujino** [comment]
As filed in https://github.com/flutter/flutter/issues/108535, the same root cause could surface as an `XmlNodeTypeException`, or other Xml exceptions. The "fix" was to catch the super class.
